api/modules/users/views.py

```diff
@@ -415,3 +415,25 @@ def generate_verification_code(request):
         error_message = "Unable to send verification code email to user"
         return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
     return Response("Verification code sent", status=status.HTTP_200_OK)
+
+
+@api_view(['GET'])
+@permission_classes((AllowAny,))
+def confirm_verification_code(request, verification_code):
+    """
+    confirm verification code sent in the request
+    :param request:
+    :return: 404 if invalid verification code
+    :return: 200 successful
+    """
+    try:
+        pass_ver = PasswordVerification.objects.get(code=verification_code)
```

Comment:
Instead of using the verification code for getting user, use the `user` information to get verification code.
```suggestion
        pass_ver = PasswordVerification.objects.get(user=request.user)
        if pass_ver == verification_code:
            ...
```